lib/kflags/map.go

```diff
@@ -0,0 +1,73 @@
+package kflags
+
+type MapAugmenter struct {
+	args map[string]string
+	manglers []VarMangler
+}
+
+type MapModifier func(e *MapAugmenter)
+
+type MapModifiers []MapModifier
+
+func (ems MapModifiers) Apply(e *MapAugmenter) {
+	for _, em := range ems {
+		em(e)
+	}
+}
+
+// WithMapMangler sets the list of manglers to use to lookup the parameters in the map.
+func WithMapMangler(m ...VarMangler) MapModifier {
+	return func(e *MapAugmenter) {
+		e.manglers = m
+	}
+}
+
+// By default, the map augmenter looks up the literal flag name and a camel case version of it.
+//
+// For example, if the 'retry-number' flag is to be filled, 'retry-number' and 'RetryNumber'
+// are both looked up in the supplied map.
+var DefaultVarMangler = []VarMangler{SkipNamespaceRemap(JoinRemap("")), SkipNamespaceRemap(JoinRemap("", CamelRewrite))}
+
+// NewMapAugmenter returns an augmenter capable of looking up flags in a map.
+//
+// For example, supply a map like map["retry-number"] = "3", and the flag
+// "retry-number" will be set to the value of 3 depending on the VarMangler
+// configured.
+func NewMapAugmenter(args map[string]string, mods ...MapModifier) *MapAugmenter {
+	augmenter := &MapAugmenter{
+		args: args,
+		manglers: DefaultVarMangler,
+	}
+
+	MapModifiers(mods).Apply(augmenter)
+	return augmenter
+}
+
+// VisitCommand implements the VisitCommand interface of Augmenter. In AssetAugmenter, it is a noop.
```

Comment:
For readability and compile time checks, could you add a type assertion?
`var _ Augmenter = &MapAugmenter{}`